src/java.desktop/macosx/native/libawt_lwawt/awt/CClipboard.m

```diff
@@ -174,6 +174,39 @@ - (BOOL) checkPasteboardWithoutNotification:(id)application {
 JNI_COCOA_EXIT(env);
 }
 
+/*
+ * Class:     sun_lwawt_macosx_CClipboard
+ * Method:    writeFileObjects
+ * Signature: ([B)V
+*/
+JNIEXPORT void JNICALL Java_sun_lwawt_macosx_CClipboard_writeFileObjects
+(JNIEnv *env, jobject inObject, jbyteArray inBytes)
+{
+    CHECK_NULL(inBytes);
+
+JNI_COCOA_ENTER(env);
+    jint nBytes = (*env)->GetArrayLength(env, inBytes);
+    jbyte *rawBytes = (*env)->GetPrimitiveArrayCritical(env, inBytes, NULL);
+    CHECK_NULL(rawBytes);
+    NSMutableArray *formatArray = [NSMutableArray arrayWithCapacity:2];
+    int i = 0;
+    for (int j = 0; j < nBytes; j++) {
+        if (rawBytes[j] == 0) {
+            NSString *path = [NSString stringWithUTF8String:(const char*)(rawBytes + i)];
+            NSURL *fileURL = [NSURL fileURLWithPath:path];
+            [formatArray addObject:fileURL.absoluteURL];
+            i = j + 1;
+        }
+    }
```

Comment:
I want to be sure I understand the intent of the above code .. it seems to indicate that the data is an arbitrary number of null terminated file names, and this code supports transferring them as an array of file URLs.
Right ? 
Any particular reason the initial array capacity is 2 ?

Reply:
Yes, that's right. Concerning the initial size, there is no particular reason to use constant 2 as an initial capacity. This factory method is used for convenience (to put NSMutableArray immediately into the autorelease pool).